pkg/agent/netpol/netpol.go

```diff
@@ -44,15 +51,37 @@ func Run(ctx context.Context, nodeConfig *config.Node) error {
 		return err
 	}
 
+	krConfig := options.NewKubeRouterConfig()
+	krConfig.ClusterIPCIDR = nodeConfig.AgentConfig.ServiceCIDR.String()
+	krConfig.NodePortRange = strings.ReplaceAll(nodeConfig.AgentConfig.ServiceNodePortRange.String(), "-", ":")
+	krConfig.HostnameOverride = nodeConfig.AgentConfig.NodeName
+	krConfig.MetricsEnabled = false
+	krConfig.RunFirewall = true
+	krConfig.RunRouter = false
+	krConfig.RunServiceProxy = false
+
 	stopCh := ctx.Done()
+	healthCh := make(chan *healthcheck.ControllerHeartbeat, healthControllerChannelLength)
+
+	// We don't use this WaitGroup, but kube-router components require it.
+	var wg sync.WaitGroup
+
 	informerFactory := informers.NewSharedInformerFactory(client, 0)
 	podInformer := informerFactory.Core().V1().Pods().Informer()
 	nsInformer := informerFactory.Core().V1().Namespaces().Informer()
 	npInformer := informerFactory.Networking().V1().NetworkPolicies().Informer()
 	informerFactory.Start(stopCh)
 	informerFactory.WaitForCacheSync(stopCh)
 
-	npc, err := NewNetworkPolicyController(client, nodeConfig, podInformer, npInformer, nsInformer, &sync.Mutex{})
+	// Start kube-router healthcheck server. Netpol requires it
+	hc, err := healthcheck.NewHealthController(krConfig)
+	if err != nil {
+		return err
+	}
+	wg.Add(1)
+	go hc.RunCheck(healthCh, stopCh, &wg)
```

Comment:
The controller handles the waitgroup decrement?

Reply:
Yes! https://github.com/cloudnativelabs/kube-router/blob/master/pkg/healthcheck/health_controller.go#L186